cmd/kops/create_cluster.go

```diff
@@ -734,10 +734,16 @@ func RunCreateCluster(ctx context.Context, f *util.Factory, out io.Writer, c *Cr
 		}
 
 		if autoloadSSHPublicKeys {
-			// Load from default location, if found
-			sshPublicKeyPath := "~/.ssh/id_rsa.pub"
-			c.SSHPublicKeys, err = loadSSHPublicKeys(sshPublicKeyPath)
-			if err != nil {
+			// Load from default locations, if found
+			sshPublicKeyPaths := []string{
+				"~/.ssh/id_rsa.pub",
+				"~/.ssh/id_ed25519.pub",
+			}
+			for _, sshPublicKeyPath := range sshPublicKeyPaths {
+				c.SSHPublicKeys, err = loadSSHPublicKeys(sshPublicKeyPath)
+				if err == nil {
+					break
+				}
 				// Don't wrap file-not-found
 				if os.IsNotExist(err) {
 					klog.V(2).Infof("ssh key not found at %s", sshPublicKeyPath)
```

Comment:
What do you think about using `multierr` in this case?
```suggestion
			var merr error
			for _, sshPublicKeyPath := range sshPublicKeyPaths {
				c.SSHPublicKeys, err = loadSSHPublicKeys(sshPublicKeyPath)
				if err == nil {
					break
				}
				// Don't wrap file-not-found
				if os.IsNotExist(err) {
					klog.V(2).Infof("ssh key not found at %s", sshPublicKeyPath)
				} else {
					merr = multierr.Append(merr, err)
				}
			}
			if err != nil && len(c.SSHPublicKeys) == 0 {
				return nil, fmt.Errorf("error reading SSH key file %q: %v", sshPublicKeyPath, merr))
			}
```

Reply:
I didn't know `multierr`, I followed your suggestion but modified it a little because sshPublicKeyPath was out of scope at the end of the loop and the function returns only an error.

Reply:
It will be easier with Go v1.20:
https://github.com/golang/go/issues/53435